plugins/woocommerce/client/blocks/assets/js/blocks/featured-items/featured-product/deprecated.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { BlockInstance, createBlock } from '@wordpress/blocks';
+import { InnerBlocks } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+
+interface BlockAttributes {
+	showDesc?: boolean;
+	showPrice?: boolean;
+	[ key: string ]: unknown;
+}
+
+// Version 1: Migration from legacy showDesc/showPrice attributes to inner blocks
+const v1 = {
+	attributes: {
+		...metadata.attributes,
+		showDesc: {
+			type: 'boolean',
+			default: true,
+		},
+		showPrice: {
+			type: 'boolean',
+			default: true,
+		},
+	},
+	save: () => <InnerBlocks.Content />,
+	isEligible: ( attributes: BlockAttributes ) => {
+		// If the block has showDesc or showPrice attributes are not explicitly set to false,
+		// it's a legacy block and it should be migrated to use inner blocks instead.
+		return attributes.showDesc !== false || attributes.showPrice !== false;
+	},
+	migrate: ( attributes: BlockAttributes, innerBlocks: BlockInstance[] ) => {
+		const { showDesc, showPrice, ...otherAttributes } = attributes;
+
+		innerBlocks.unshift(
+			createBlock( 'core/post-title', {
+				level: 2,
+				isLink: false,
+				textAlign: 'center',
+				__woocommerceNamespace:
+					'woocommerce/product-collection/product-title',
+			} )
+		);
+
+		// We check if these legacy attributes are explicitly set to false, because
+		// the default value is true (i.e. `undefined` meant `true`).
+		if ( showPrice !== false ) {
+			innerBlocks.push( createBlock( 'woocommerce/product-price' ) );
+		}
+
+		if ( showDesc !== false ) {
+			innerBlocks.push(
```

Comment:
Same here: :slightly_smiling_face: 
```suggestion
			innerBlocks.unshift(
```

Reply:
See above.